mapbox/libjava-services/src/main/java/com/mapbox/services/api/staticimage/v1/models/Marker.java

```diff
@@ -0,0 +1,291 @@
+package com.mapbox.services.api.staticimage.v1.models;
+
+import com.mapbox.services.Constants;
+import com.mapbox.services.api.ServicesException;
+import com.mapbox.services.commons.models.Position;
+
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Mapbox Static Image API marker overlay. Building this object allows you to place a marker on top or within
+ * your static image. The marker can either use the default marker (though you can change it's color and size) or you
+ * have the option to also pass in a custom marker icon using it's url.
+ *
+ * @since 2.1.0
+ */
+public class Marker {
+
+  private String marker;
+
+  /**
+   * A Marker constructor
+   *
+   * @param builder a Marker builder.
+   * @since 2.1.0
+   */
+  public Marker(Builder builder) {
+
+    if (builder.getPrecision() > 0) {
+      String pattern = "0." + new String(new char[builder.getPrecision()]).replace("\0", "0");
+      DecimalFormat df = (DecimalFormat) DecimalFormat.getInstance(Locale.US);
```

Comment:
Instead of having custom code here for formatting, let's extend the code in https://github.com/mapbox/mapbox-java/pull/424 for this use case?